### ISSUE TITLE
Deprecate `didCreateRootView`

### ISSUE THREAD

**Issue by chancancode** [emberjs/ember.js#14136]
This was added for fastboot, but is no longer used. It also leaks the "top-level view" which is not really a thing anymore.

**Comment by AKST** [comment]
How does one deprecate a feature, what is involved in deprecating a feature? Are the existing examples of how other features have been deprecated?

**Comment by AKST** [comment]
# First time contributor questions


## Assumptions

These are my assumptions as a first time contributor after looking over the `contributing.md` and looking at some related deprecation PRs

- I'm assuming master has the latest changes so I branch off there for all my changes
- I'm assuming my PR will be against the master branch
- I'm assuming the until property of the deprecation is `3.0.0`

## Id for deprecation

Is `ember-application.app-instance.did-create-root-view` a good id for the deprecation warning? I noticed most of the existing method deprecations have dashed separated names, is this an appropriate id for `ApplicationInstance.didCreateRootView`?

## Url for deprecation?

Other PRs did not show any changes for changes to documentation linked to by these deprecation warnings, is there an existing link this deprecation should link to?

## Message for deprecation

> "Using `ApplicationInstance.didCreateRootView` is deprecated"

Is this an adequate message, or is there an alternative api this message should suggested to use in this features place? (like the deprecation for `ApplicationInstance.container.lookup`).

## Tests

It appears this method is called at boot up for a majority of tests, so tons of tests break unsurprisingly. Is the best approach to add an `expectDeprecation` clause every where this is the case? Should I write tests specific to this deprecation?

**Comment by AKST** [comment]
There is one place in the code base this methods is invoked, and that's [in the router][call]. Should I update the call to `ApplicationInstance.didCreateRootView` in the router to be the esstenially what that call does (as it's one line)?

[call]: https://github.com/emberjs/ember.js/blob/beta/packages/ember-routing/lib/system/router.js#L326

So this

```javascript
    //  ...
      let instance = owner.lookup('-application-instance:main');
      instance.didCreateRootView(this._toplevelView);
    // ...
```

Becomes this

```javascript
    //  ...
      let instance = owner.lookup('-application-instance:main');
      this._toplevelView.appendTo(instance.rootElement);
    // ...
```

And now all the tests pass, and I'm assuming I should create a test that checks if the deprecation warning is fired for directly calling `didCreateRootView`?

**Comment by AKST** [comment]
Lastly is there any eduquitte around the yarn.lock file? do I just not add to the commit?

**Comment by btecu** [comment]
Unless you've updated anything in `packages.json`, you should discard any `yarn.lock` changes.

**Comment by AKST** [comment]
Don't worry I didn't commit it, I'm starting to think it's my nose version

**Comment by kratiahuja** [comment]
It's actually [used in FastBoot](https://github.com/ember-fastboot/ember-cli-fastboot/blob/master/app/instance-initializers/browser/clear-double-boot.js#L18) yet.

**Comment by rwjblue** [comment]
Closed by #17148
